ash/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![deny(clippy::use_self)]
+#![warn(/* trivial_casts, */ trivial_numeric_casts)]
```

Comment:
I enabled `trivial_casts` here to find many violations "by hand". Unfortunately we have many false-positives where `rustc` thinks a coercion can take place because it firsts casts a borrow to a raw pointer of the same type, then casts that pointer to a different type. But that coercion of the first cast cannot happen without having the type explicit :(

```rust
warning: trivial cast: `&'a mut T` as `*mut T`
   --> ash/src/extensions/experimental/amd.rs:697:28
    |
697 |             let next_ptr = next as *mut T as *mut BaseOutStructure;
    |                            ^^^^^^^^^^^^^^
    |
    = help: cast can be replaced by coercion; this might require a temporary variable
```

Reply:
It might be helpful to make more explicit use of the `cast` method on pointers. This should be able to replace most uses of `as` entirely. For example, `let next_ptr = <*mut T>::cast::<BaseOutStructure>(next)`.